include/xtensor/xsort.hpp

```diff
@@ -410,8 +410,8 @@ namespace xt
         return ev;
     }
 
-    template <class E, std::size_t N, class R = typename detail::flatten_sort_result_type<E>::type>
-    inline R partition(const xexpression<E>& e, const std::size_t (&kth_container)[N], placeholders::xtuph tag)
+    template <class E, class I, std::size_t N, class R = typename detail::flatten_sort_result_type<E>::type>
+    inline R partition(const xexpression<E>& e, const I(&kth_container)[N], placeholders::xtuph tag)
```

Comment:
It looks like the partition and argpartition functions matching this signature (with argument `const I(&kth_container)[N]`) need an `std::initializer_list<I>` implementation for `X_OLD_CLANG` as the clang3.6 tests failed on travis.

Reply:
Exactly! Thanks for the comment! It's fixed now ...  if you want to give it a review, feel free. I hope we'll merge it by today

Reply:
Great! Thanks so much!

Sorry I couldn't test myself but I don't have an older clang set up and it seems I can't have a fork of the main repo and your fork at the same time.

Reply:
No worries. We also only mostly use travis for testing older clang's

Reply:
template parmater `R` should be in second position.